### PR TITLE
TA-5373 Cypress Exercise

### PR DESCRIPTION
**Pull request by krumkarapetrov-taulia**
Cypress exercise PR

### CODE REVIEW

**Comment by krumkarapetrov-taulia** [comment]
@krasimiryanchev-taulia PR for the Cypress exercise as requested on our last meeting.

File: cypress/e2e/1-qe-interview-task/Exercise.cy.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        cy.visit('/exercise');
+        let count = 0;
+        cy.get('.table-wrapper tbody')
+            .then($tbody => {
+                 count = $tbody.find('tr').length
+                cy.get('div > .tau-button')
+                    .should('be.visible')
+                    .click();
+                cy.get('[data-testid="modal-header"]')
+                    .should("contain", 'Example Modal');
```
Comment (krasimiryanchev-taulia) [inline cypress/e2e/1-qe-interview-task/Exercise.cy.js:38]: Please use single quotes consistently like here:
`"contain" => 'contain'`

File: cypress/e2e/1-qe-interview-task/BadLink.cy.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+describe('BadLink',() =>{
+
+    it('should navigate to the bad link and get a 404 error',  () => {
+
```
Comment (krasimiryanchev-taulia) [inline cypress/e2e/1-qe-interview-task/BadLink.cy.js:4]: Avoid new lines after describe and around the `it` block

File: cypress/e2e/1-qe-interview-task/BadLink.cy.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+describe('BadLink',() =>{
```
Comment (krasimiryanchev-taulia) [inline cypress/e2e/1-qe-interview-task/BadLink.cy.js:1]: Have a space after `,` and after `=>` like this:
`'BadLink', () => {`

File: cypress/e2e/1-qe-interview-task/BadLink.cy.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    it('should navigate to the bad link and get a 404 error',  () => {
```
Comment (krasimiryanchev-taulia) [inline cypress/e2e/1-qe-interview-task/BadLink.cy.js:3]: Replace the two spaces with only one like this:
`error', ()`

File: cypress/e2e/1-qe-interview-task/BadLink.cy.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+        cy.visit('/');
+        cy.visit('/omg-why-is-this-even-a-url-this-is-totally-wrong');
+        cy.get('p').should('contain','We can\'t seem to find the page you\'re looking for.');
```
Comment (krasimiryanchev-taulia) [inline cypress/e2e/1-qe-interview-task/BadLink.cy.js:7]: Add space after `,` like this:
`'contain', 'We `

File: cypress/e2e/1-qe-interview-task/Exercise.cy.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+    it('should check that submit button is not active when one of required fields is empty', ()=> {
+
+        cy.visit('/');
+        cy.contains('Welcome').should('exist');
+        cy.visit('/exercise');
+        cy.contains('Full Name').should('exist');
+        cy.contains('Email').should('exist');
+        cy.contains('Role').should('exist');
+        cy.fillExerciseForm(user,email,'{backspace}');
```
Comment (krasimiryanchev-taulia) [inline cypress/e2e/1-qe-interview-task/Exercise.cy.js:76]: Have all the parameters separated by spaces like:
`(user, email, '{backspace}')`

**Review by krasimiryanchev-taulia — CHANGES_REQUESTED** [review]
The logic behind this PR looks great and works great! 💯

My comments are only related to the code formatting.
Please check those two links we follow as guidelines about formatting:
https://github.com/airbnb/javascript
https://airbnb.io/javascript/react

Please use the IDE auto-format feature to correct most of those comments!
If using IntelliJ - Prettier should be already bundled.
If using VSCode - please install and configure the appropriate extension.

**Review by krasimiryanchev-taulia — APPROVED** [review]
🏆

**Pull request closed without merging**